duckduckgo/iOS
# internal feature flagging

Task/Issue URL:
Tech Design URL: https://app.asana.com/0/481882893211075/1202291296379036/f
CC:

**Description**:
Adds very basic feature flagging for internal users. A user is permanently marked as internal if they successfully authenticate through login.duckduckgo.com. isInternalUser also always returns true for debug builds

Also makes the debug menu only appear if the user is internal



**Steps to test this PR**:
Note, you may need to test on an actual device so you don't get out of date software errors when trying to authenticate
1. Go to settings, you should see the debug menu option, check that it works
1. On settings, also try pressing on the version number, it should no longer take you to the debug menu.
-----
1. In "DefaultFeatureFlagger.isInternalUser, so you can test the next bit comment out:
```
if isDebugBuild {
        return true
 }
```
2. Go to settings again, debug menu option should no longer appear, and the version number still shouldn't do anything
---
1. With that code still commented out, go to a number of sites, including DDG sites (but *not* login.duckduckgo.com)
1. Go to settings again, debug menu option shouldn't be there
---
1. Go to "login.duckduckgo.com/patestssucceeded". It should show an error page.
2. Go to settings again, there should still be no debug menu. 
---
1. Go to "login.duckduckgo.com" but don't complete authentication
2. Go to settings again, there should still be no debug menu. 
---
1. Now finally go to "login.duckduckgo.com"and complete authentication
2. Go to settings and check the debug menu appears


**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
